pkg/api/ui_handler.go

```diff
@@ -16,11 +17,19 @@ import (
 	"github.com/treeverse/lakefs/webui"
 )
 
-func NewUIHandler(gatewayDomains []string) http.Handler {
-	content, err := fs.Sub(webui.Content, "dist")
-	if err != nil {
-		// embedded UI content is missing
-		panic(err)
+func NewUIHandler(gatewayDomains []string, fsDir *string) http.Handler {
+	var (
+		content fs.FS
+		err     error
+	)
+	if fsDir == nil {
+		content, err = fs.Sub(webui.Content, "dist")
+		if err != nil {
+			// embedded UI content is missing
+			panic(err)
+		}
+	} else {
+		content = os.DirFS(*fsDir)
```

Comment:
I don't know if there is a difference (from the code it looks like both works) - usually the classic implementation uses `http.Dir`